src/ipc/handler.c

```diff
@@ -663,11 +663,19 @@ static int ipc_pm_gate(uint32_t header)
 
 	IPC_COPY_CMD(pm_gate, _ipc->comp_data);
 
+	/* pause dma trace firstly if needed */
+	if (pm_gate.flags & SOF_PM_NO_TRACE)
+		trace_off();
+
 	if (pm_gate.flags & SOF_PM_PPG)
 		pm_runtime_disable(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID);
 	else
 		pm_runtime_enable(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID);
 
+	/* resume dma trace if needed */
+	if (!(pm_gate.flags & SOF_PM_NO_TRACE))
+		trace_on();
```

Comment:
@mmaka1 a couple of questions:
1. The condition !(pm_gate.flags & SOF_PM_NO_TRACE) will always be true isnt it?
2. In the kernel S0ix implementation, we now have included the feature to request trace DMA to be kept enabled when the DSP enters D0i3 with the system in S0. Just wanted to make sure the FW supports this.
1. Secondly,

Reply:
Correction: sorry I misread the ! As a ~. The condition is good.